### PR TITLE
English Translation of ai-edu/A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-搭建深度神经网络框架.md

### PR DESCRIPTION
**Pull request by marko-polo-cheno**

### CODE REVIEW

File: A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+<!--Copyright © Microsoft Corporation. All rights reserved.
+  适用于[License](https://github.com/Microsoft/ai-edu/blob/master/LICENSE.md)版权许可-->
+
+# Chapter 14 Building a Deep Neural Network Framework
+
+## 14.0 Deep Neural Network Framework Design
+
+### 14.0.1 Function/Mode Analysis
+
+Compared to code of the three-layer neural network in Chapter 12, we can see lots of repetition, such as in the forward propagation:
```
Comment (marko-polo-cheno) [inline A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md:10]: forward propagation vs forward calculation?

File: A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    dB2 = np.sum(dZ2, axis=1, keepdims=True)
+    # layer 1
+    dZ1 = np.dot(W2.T, dZ2) * A1 * (1-A1)   #sigmoid
+    dW1 = np.dot(dZ1, X.T)
+    dB1 = np.sum(dZ1, axis=1, keepdims=True)
+    ...
+```
+The pattern of each layer is also very similar: calculate `dZ` of a layer, and then calculate `dW` and `dB` according to `dZ`.
+
+As the three-layer network has one more layer than the two-layer network, the four stages of initialization, forward propagation, backward propagation, and updating parameters is slightly different. In addition, in the previous chapters, in order to achieve some auxiliary functions, we have written a lot of classes. So, now you can build a mini-framework for deep learning.
```
Comment (marko-polo-cheno) [inline A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md:50]: forward propagation vs forward calculation?

File: A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+#### NeuralNet
+
+First, a `NeuralNet` class is needed to package the basic neural network structure and functions:
+
+- `Layers` - the containers of each layer of the neural network, maintains a list in the order of addition
+- `Parameters` - basic parameters, including common parameters and hyperparameters
+- `Loss Function` - provides the function of calculating loss function value, storing historical records, and finally drawing it
+- `LayerManagement()` - addd a neural network layer
+- `ForwardCalculation()` - calls the forward propagation method on each layer
```
Comment (marko-polo-cheno) [inline A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md:68]: forward propagation vs forward calculation?

File: A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+- Dropout Layer
+- Batch Norm Layer
+
+In the future, it will also include:
+
+- Convolution Layer
+- Max Pool Layer
+
+Each Layer includes the following basic methods：
+ - `ForwardCalculation()` - calls the forward propagation method of this layer
```
Comment (marko-polo-cheno) [inline A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md:92]: forward propagation vs forward calculation?

**Comment by marko-polo-cheno** [comment]
Is forward propagation the same as forward calculation?

File: A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md (hunk 3 above)
Comment (xiaowuhu) [reply A-基础教程/A2-神经网络基本原理/en-us/Step7 - DNN/14.0-Building a DNN framework.md:68]: should be calculation

**Review by xiaowuhu — APPROVED** [review]
